### DISCUSSION TITLE
Minio bad request: Invalid header name when using fput_object

### DISCUSSION THREAD

**Discussion by yichong96** [minio/minio#13645] [Q&A] [answered]
I am using python SDK client to upload an image into my minio localhost server. I have not been successful in using the api. This is the python code that I have.

```
from minio import Minio
from minio.error import S3Error
import json
import numpy as np
import argparse
import glob

def main():
    client = Minio("localhost:9000",
                access_key="XXXX",
                secret_key="XXXX",
                secure=False)
    
    parser = argparse.ArgumentParser()
    parser.add_argument("image_filepath", type=str)
    parser.add_argument("-b", "--bucket_name", type=str)
    parser.add_argument("-i", "--operation", required=True)
    FLAGS = parser.parse_args()

    # check if bucket exist
    myBucket = FLAGS.bucket_name
    found = client.bucket_exists(myBucket)
    if found:
        print("Found: " + myBucket)
    else:
        client.make_bucket(myBucket)
        print("Creating " + myBucket +  " bucket")
        
    if FLAGS.operation == "read":
        objects = client.list_objects(myBucket)
        data = []
        for obj in objects:
            response = client.get_object(myBucket, obj.object_name)
            print(response.getheaders())
            print(response.read())
            #deserialized_bytes = np.frombuffer(response, dtype=np.float32)
            #print(deserialized_bytes)
            #np_array = np.reshape(deserialized_bytes, newshape=(1, 2048))
            #print(np_array)
            #data.append(response)
    
    if FLAGS.operation == "write":
        # upload operation
        count = 0
        for file in glob.glob(FLAGS.image_filepath + "/*.jpg"):
            metadata = {"file number": count}
            object_name = file.split("/")[-1]
            print(file)
            result = client.fput_object(FLAGS.bucket_name, object_name, file, metadata=metadata)
            print("object name added: {}".format(result.object_name))
            count += 1
            

if __name__ == "__main__":
    main()
```

Im not sure why I am encountering an error in `result = client.fput_object(FLAGS.bucket_name, object_name, file, metadata=metadata)`. The error logs are as below:

> Traceback (most recent call last):
>   File "minio_client.py", line 56, in <module>
>     main()
>   File "minio_client.py", line 49, in main
>     result = client.fput_object(FLAGS.bucket_name, object_name, file, metadata=metadata)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/minio/api.py", line 988, in fput_object
>     tags=tags, retention=retention, legal_hold=legal_hold,
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/minio/api.py", line 1758, in put_object
>     raise exc
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/minio/api.py", line 1713, in put_object
>     bucket_name, object_name, part_data, headers,
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/minio/api.py", line 1577, in _put_object
>     no_body_trace=True,
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/minio/api.py", line 406, in _execute
>     no_body_trace=no_body_trace,
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/minio/api.py", line 309, in _url_open
>     response.data.decode() if response.data else None,
> minio.error.InvalidResponseError: non-XML response from server; Response code: 400, Content-Type: text/plain; charset=utf-8, Body: 400 Bad Request: invalid header name

Any help is greatly appreciated !

**Comment by harshavardhana** [comment, marked as answer]
> Im not sure why I am encountering an error in `result = client.fput_object(FLAGS.bucket_name, object_name, file, metadata=metadata)`. The error logs are as below:
> 
> >

Please print the metadata key and value.

**Reply by yichong96** [reply]
the metadata follows the format in the code `{'file number': i}` where `i` is an integer

**Reply by harshavardhana** [reply]
> the metadata follows the format in the code `{'file number': i}` where `i` is an integer

you cannot have `space` in the metadata name that is not allowed HTTP header name.

**Reply by yichong96** [reply]
I see thanks :-) for your help ! >.<
